daemon.go

```diff
@@ -59,12 +60,27 @@ func (daemon menderDaemon) StopDaemon() {
 
 func (daemon *menderDaemon) Run() error {
 	// figure out the state
-	switch daemon.GetState() {
-	case MenderFreshInstall:
-		//do nothing
-	case MenderRunningWithFreshUpdate:
-		daemon.CommitUpdate()
+	for {
+		switch daemon.TransitionState() {
+		case MenderStateRunningWithFreshUpdate:
+			daemon.CommitUpdate()
```

Comment:
I know this was a part of refactoring, but `CommitUpdate()` in fact can return error and we should check for it. something to remember to add in future.